src/GeneralStateTestsFiller/stCreateTest/CreateOOGafterMaxCodesizeFiller.yml

```diff
@@ -0,0 +1,337 @@
+          // create contracts with max codesize in loop
+          for { let i := 0 } lt(i, contract_count) { i := add(i, 1) }
+          {
+              let address_created := create(0, 0, initcode_size)
+              mstore( add(initcode_size, mul(i, 32)), address_created )
+          }
+          if gt(should_oog, 0) {
+            invalid()
+          }
+          return(initcode_size, mul(contract_count, 32))
+        }
+      nonce: 1
+      storage: {}
+
+    00000000000000000000000000000000000c0dea:
+      balance: 0
+      code: |
+        :yul {
+
+          // Get the amount of contracts to create on this level
+          let delegate_contract_count := calldataload(4)
+
+          // Get the amount of contracts to create on the sub level call
+          let subcall_contract_count := calldataload(36)
+
+          // Get whether the subcall should oog
+          let subcall_oog := calldataload(68)
+
+          // Get count of contracts to call to self-destruct
+          let selfdestruct_count := calldataload(100)
+
+          // Call for contract creation
+          mstore(0, delegate_contract_count)
+          mstore(32, 0)
+          let retcode := delegatecall(div(gas(), 2), 0x00000000000000000000000000000000000c0deb, 0, 64, 64, mul(delegate_contract_count, 32))
+
+          // Call for OOG contract creation
+          mstore(0, subcall_contract_count)
+          mstore(32, subcall_oog)
+          retcode := call(div(gas(), 2), 0x00000000000000000000000000000000000c0deb, 0,	0, 64, add(64, mul(delegate_contract_count, 32)), mul(subcall_contract_count, 32))
+
+          for { let i := 0 } lt(i, selfdestruct_count) { i := add(i, 1) }
+          {
+              let contract_addr := mload(add(64, mul(i, 32)))
+              pop(call(sub(gas(), 1000), contract_addr, 0, 0, 0, 0, 0))
+          }
+
+        }
+      nonce: 1
+      storage: {}
+
+    a94f5374fce5edbc8e2a8697c15331677e6ebf0b:
+      balance: '0x0ba1a9ce0ba1a9ce'
+      code: '0x'
+      nonce: '0'
+      storage: {}
+
+
+  transaction:
+    data:
+    # Parameters:
+    #  1) Number of contracts to create using delegate call (On c0dea)
+    #  2) Number of contracts to create using call (On c0deb)
+    #  3) Whether the second call should oog
+    #  4) Number of contracts to call to selfdestruct
+    # - Low number of contracts (10)
+    - :abi f(uint,uint,bool,uint) 0x00 0x0a 0x01 0x00
+    - :abi f(uint,uint,bool,uint) 0x0a 0x0a 0x01 0x00
+    - :abi f(uint,uint,bool,uint) 0x0a 0x0a 0x00 0x0e
+    # - High number of contracts (250)
+    - :abi f(uint,uint,bool,uint) 0x00 0xfa 0x01 0x00
+    - :abi f(uint,uint,bool,uint) 0xfa 0xfa 0x01 0x00
+    - :abi f(uint,uint,bool,uint) 0xfa 0xfa 0x00 0x01ee
+    gasLimit:
+    - 0x100000000
+    gasPrice: '10'
+    nonce: '0'
+    to: 00000000000000000000000000000000000c0dea
+    secretKey: "45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8"
+    value:
+    - 0
+
+  expect:
+
+    - indexes:
+        data: !!int 0
```

Comment:
can you comment what happens in this case? which scenario

Reply:
This is scenario:
`A stores 1:1, A calls B, B create contracts with maxcodesize 10 times (no oog upto here), then B oogs the call. creations reverted. But A still stores 1:1 as it is the subcall that OOGed`

We are passing the following parameters:
1) 0x00, the top level contract A creates zero contracts using delegateCall (to reuse the contract creating code)
2) 0x0a, the 2nd level contract B creates 10 contracts
3) 0x01, the contract B will issue an invalid opcode after creating the contracts to induce the OOG.
4) 0x00, contract A calls zero contracts to perform self-destruct (since the Contract B reverted anyway).

Reply:
I mean in the test

Reply:
I added comments to each contract (0xc0dea, 0xc0deb), also comments to each test case.
Let me know if this makes it more readable.

Reply:
wait, if you do sstore inside delegatecall (which actually I meant not a delegatecall there) that goes oog
why would sstore be still valid?

Reply:
ah, first you do another delegate subcall I see.
so the sstore here basically means that the first delegate call worked. 

I use sstore to signal that contract code has actually worked. 
so you can even put sstore (to a different index) in each contract that you call (at the end of the call) to verify that those have worked correctly